### PR TITLE
Adds proxy auth support to pulp_repo module

### PR DESCRIPTION
**Pull request by dcode**
##### SUMMARY
Added support for the pulp_repo module to allow for authentication to proxies. Adds the `proxy_username` and `proxy_password` options to the REST API calls.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
pulp_repo module

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/Users/<scrubbed>/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.4.0.0/libexec/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

### CODE REVIEW

**Review by sysadmind — CHANGES_REQUESTED** [review]
Looks good. Just needs the pep8 updates that the Ansible commented.

**Comment by dcode** [comment]
Working on it. I'll get it updated today.

File: lib/ansible/modules/packaging/os/pulp_repo.py [hunk 1]
```diff
@@ -100,6 +112,12 @@
     description:
       - Repo plugin type to use (i.e. C(rpm), C(docker)).
     default: rpm
+  repoview:
+    description:
+      - Enables repoview of published repository to improve browsing experience
```
Comment (sysadmind) [inline lib/ansible/modules/packaging/os/pulp_repo.py]: I think that `enable` indicates that this option can only enable the repoview and not delete it, but it looks like all of the code will still work to remove it as well (and it should). Looking at other modules, the phrasing is more often along the lines of `Whether to generate repoview files` or `Manage distributor repoview property`

**Review by sysadmind — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/packaging/os/pulp_repo.py (hunk 1 above)
Comment (dcode) [reply lib/ansible/modules/packaging/os/pulp_repo.py]: Sounds good. Pushed a fix.

**Review by sysadmind — APPROVED** [review]
Looks good. Thanks for the contribution!

shipit

**Comment by sysadmind** [comment]
shipit

**Comment by dcode** [comment]
Sorry...I tested this in a sterile environment with a newer pulp installation and the `repoview` option failed via the API call without the `generate_sqlite` option. So, I added that option and now it works. The CLI tool automatically sets this, so using `repoview: True` will automatically set `generate_sqlite: True` in the module now.

**Comment by resmo** [comment]
code LGTM
shipit

**Comment by gthieleb** [comment]
This commit fixes the authenticaton for `pulp_repo` module against cooperate proxy and works in our environment with RHEL7 and pulp 2.14.0 [1].

Please ship that fix with an upcoming ansible version.

[1] 
```
$ rpm -qa|grep pulp|grep rpm
pulp-rpm-admin-extensions-2.14.0-1.el7.noarch
pulp-rpm-plugins-2.14.0-1.el7.noarch
python-pulp-rpm-common-2.14.0-1.el7.noarch
pulp-rpm-consumer-extensions-2.14.0-1.el7.noarch
pulp-rpm-handlers-2.14.0-1.el7.noarch
pulp-rpm-yumplugins-2.14.0-1.el7.noarch
```

**Comment by dcode** [comment]
I was trying to rebase this, but clearly messed up. Working on fixing it. I might open a new PR with my squashed changes if I can figure it out.

**Pull request closed without merging**